packages/admin/admin-generator/src/commands/generate/generateForm/generateFormValues.ts

```diff
@@ -168,17 +242,46 @@ function generateFormValuesToGqlInputFromTree(tree: FormValuesConfigTree, dataOb
             }
         } else if (value.config) {
             if (value.config.formValueToGqlInputCode) {
-                code += `${key}: ${value.config.formValueToGqlInputCode}, `;
+                code += `${key}: ${value.config.formValueToGqlInputCode.replaceAll("$fieldName", `${dataObject}.${key}`)}, `;
             }
         }
     }
     return code;
 }
 
-export function generateFormValuesToGqlInput({ formValuesConfig }: { formValuesConfig: GenerateFieldsReturn["formValuesConfig"] }) {
-    const tree = formValuesConfigToTree(formValuesConfig);
-    return `const output = {
-        ...formValues,
-        ${generateFormValuesToGqlInputFromTree(tree, "formValues")}
-    };`;
+function hasChildDestructTree(tree: FormValuesConfigTree): boolean {
+    return Object.values(tree).some((childValue) => {
+        if (childValue.config?.destructFromFormValues) {
+            return true;
+        }
+        if (Object.keys(childValue.children).length > 0) {
+            return hasChildDestructTree(childValue.children);
+        }
+        return false;
+    });
+}
+
+export function generateFormValuesToGqlInput({
+    formValuesConfig,
+    gqlIntrospection,
+    gqlType,
+}: {
+    formValuesConfig: GenerateFieldsReturn["formValuesConfig"];
+    gqlIntrospection: IntrospectionQuery;
+    gqlType: string;
+}) {
+    const tree = formValuesConfigToTree({ formValuesConfig, gqlIntrospection, gqlType });
+    const hasChildDestruct = hasChildDestructTree(tree);
+    const code = generateFormValuesToGqlInputFromTree(
+        tree,
+        hasChildDestruct ? "formValuesRest" : "formValues",
```

Comment:
Why do we need this differentiation?

Reply:
because ...Rest is only used when it's actually a rest like here:
`const handleSubmit = async ({ dimensionsEnabled, ...formValuesRest }: FormValues, form: FormApi<FormValues>, event: FinalFormSubmitEvent) => {`

when we don't need destruction the following code is generated:
`const handleSubmit = async (formValues: FormValues, form: FormApi<FormValues>, event: FinalFormSubmitEvent) => {`